### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by p0psicles**

### CODE REVIEW

File: adba/aniDBAbstracter.py [hunk 1]
```diff
@@ -17,15 +17,16 @@
 # along with aDBa.  If not, see <http://www.gnu.org/licenses/>.
 
 import logging
-import os
+from os import makedirs, path
```
Comment (medariox) [inline adba/aniDBAbstracter.py]: I would import os and use os.makedirs, os.path (like the rest).

File: adba/aniDBfileInfo.py [hunk 2]
```diff
@@ -98,22 +99,70 @@ def get_file_size(path):
     return size
 
 
-def read_anidb_xml(file_path):
-    if not file_path:
+def read_anidb_xml(file_path=None):
+    if not os.path.join(file_path, "animetitles.xml"):
```
Comment (medariox) [inline adba/aniDBfileInfo.py]: Joining with None will cause an exception.

File: adba/aniDBfileInfo.py [hunk 3]
```diff
@@ -98,22 +99,70 @@ def get_file_size(path):
...
     f = open(filePath, "r")
     xml_a_setree = etree.ElementTree(file=f)
     return xml_a_setree
+
+
+def _remove_file_failed(file):
+    try:
+        os.remove(file)
+    except:
+        pass
```
Comment (medariox) [inline adba/aniDBfileInfo.py]: A log message would be nice.

File: adbacli.py [hunk 4]
```diff
@@ -115,13 +113,18 @@
         connection.auth(args.username, args.password)
     except Exception as e:
         print('Exception: %s', e)
+        logging.debug('Exception Occurred: ' + str(e))
```
Comment (medariox) [inline adbacli.py]: I would use `%r` and have the occurred lowercase. Same for all the others.

File: adba/aniDBAbstracter.py [hunk 5]
```diff
@@ -80,7 +81,7 @@ def _easy_extend(initialList, item):
         if item:
             if isinstance(item, list):
                 initialList.extend(item)
-            elif isinstance(item, str):
+            elif isinstance(item, string_types):
```
Comment (medariox) [inline adba/aniDBAbstracter.py:84]: Prefer to limit to `text_type` if possible.

File: adba/aniDBAbstracter.py [hunk 6]
```diff
@@ -99,15 +100,20 @@ def add_notification(self):
...
 
 class Anime(aniDBabstractObject):
-    def __init__(self, aniDB, name=None, aid=None, tvdbid=None, paramsA=None, autoCorrectName=False, load=False):
+    def __init__(self, aniDB, name=None, aid=None, tvdbid=None, paramsA=None, autoCorrectName=False, load=False, cache_path=None):
 
         self.maper = AniDBMaper()
-        self.tvDBMap = TvDBMap()
+
+        if cache_path and not path.exists(cache_path):
+            makedirs(cache_path)
```
Comment (medariox) [inline adba/aniDBAbstracter.py]: What if makedirs fails? Missing permissions, etc. Graceful handling?

File: adba/aniDBAbstracter.py [hunk 7]
```diff
@@ -196,13 +202,8 @@ def _get_name_from_xml(self, aid, onlyMain=True):
...
         return ""
 
-    @staticmethod
-    def _read_animetitels_xml(path=None):
-        if not path:
-            path = os.path.join(os.path.dirname(os.path.abspath(__file__)), "animetitles.xml")
-
-        f = open(path, "r")
-        all_anime_xml = etree.ElementTree(file=f)
+    def _read_animetitels_xml(self):
```
Comment (medariox) [inline adba/aniDBAbstracter.py]: This function can be removed altogether and `read_anidb_xml` used directly instead.

File: adbacli.py (hunk 4 above)
Comment (labrys) [reply adbacli.py]: Also the line above does not print as you would expect.
```python
>>> e = 'test'
>>> print('Exception: %s', e)
Exception: %s test
```

File: adba/aniDBAbstracter.py (hunk 5 above)
Comment (p0psicles) [reply adba/aniDBAbstracter.py:84]: I don't know. What if item is a binary type? I'm not confident with this code, to assume it's always unicode.

File: adba/aniDBAbstracter.py (hunk 6 above)
Comment (p0psicles) [reply adba/aniDBAbstracter.py]: How would you gracefull handle this? Show an error, and reraise the exception?
I can catch it with an OSerror. But then what? It should stop doing anything after that right?

File: adbacli.py [hunk 8]
```diff
@@ -112,8 +112,8 @@
     try:
         connection.auth(args.username, args.password)
     except Exception as e:
-        print('Exception: %s', e)
-        logging.debug('Exception Occurred: ' + str(e))
+        print('Exception: %s' % e)
```
Comment (labrys) [inline adbacli.py]: This same error is in most of the exceptions.

Comment (p0psicles) [reply adbacli.py]: Aah yes i see. Ill also change those.

**Pull request merged**